lib/Core/Search/Solr/ResultExtractor.php

```diff
@@ -12,10 +16,23 @@
  */
 abstract class ResultExtractor Extends BaseResultExtractor
 {
-    public function extract($data, array $facetBuilders = [])
+    public function extract($data, array $facetBuilders = [], ?Query $query = null)
     {
         $searchResult = $this->extractSearchResult($data, $facetBuilders);
 
+        foreach ($searchResult->searchHits as $key => $searchHit) {
+            $searchResult->searchHits[$key] = new SearchHit(get_object_vars($searchHit));
+            $searchResult->searchHits[$key]->extraFields = [];
+
+            if ($query instanceof ExtraQuery || $query instanceof ExtraLocationQuery && is_array($query->extraFields)) {
```

Comment:
First two conditions need to be wrapped into parentheses, as `&&` has precedence over `||`.

Reply:
@pspanja Fixed in https://github.com/netgen/ezplatform-search-extra/pull/52/commits/24cfb1ce1450e74ea62c663b462e72b23045ceb9.